department-of-veterans-affairs/abd-vro
# bump pydantic version for Contention Classification App

## What was the problem?

- updated pydantic version in aim to avoid secrel issues requiring update at unexpected time mid-sprint

Associated tickets or Slack threads:
- https://github.com/department-of-veterans-affairs/vagov-claim-classification/issues/374

## How does this fix it?[^1]

- I used the recommend `bump-pydantic` package from the [pydantic migration guide](https://docs.pydantic.dev/2.6/migration/)
- Updated syntax for @model_validator manually
- updated requirements.txt

## How to test this PR
- run `pytest`
- test POST request to cc-app via postman, swagger docs, etc


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
